core/src/test/scala/org/apache/spark/sql/delta/DeltaSourceSuite.scala

```diff
@@ -1388,6 +1388,61 @@ class DeltaSourceSuite extends DeltaSourceSuiteBase with DeltaSQLCommandTest {
     }
   }
 
+  test("startingVersion earliest") {
+    withTempDir { dir =>
+      withTempView("startingVersionEarliest") {
+        val path = dir.getAbsolutePath
+        spark.range(0, 10).write.format("delta").save(path)
+
+        def startTestStreamingQuery(): StreamingQuery = {
+          spark.readStream
+            .format("delta")
+            .option("startingVersion", "earliest")
+            .load(path)
+            .writeStream
+            .format("memory")
+            .queryName("startingVersionEarliest")
+            .start()
+        }
+
+        var q = startTestStreamingQuery()
+        try {
+          q.processAllAvailable()
+          checkAnswer(sql("select * from startingVersionEarliest"), (0 until 10).map(Row(_)))
```

Comment:
nit: `0.until(10).map(Row.apply)` looks less "overwhelming" and "scary" :)

Reply:
I thought the current syntax looked simpler, as it resembles a mathematical range (i.e. `(0, 10)`). I also wanted to stay consistent with simliar tests in this file. If you still prefer it the other way, I'll change it.